ui/src/reducers/search.js

```diff
@@ -5,104 +5,203 @@ import {
+      );
+    case CHANGE_SEARCH_BOX_NAMESPACE:
+      return state.set('searchBoxNamespace', searchBoxNamespace);
+    case NEW_SEARCH_REQUEST:
+      return state
+        .setIn(
+          ['namespaces', namespace, 'initialAggregations'],
+          initialState.getIn(['namespaces', namespace, 'initialAggregations'])
+        )
+        .setIn(
+          ['namespaces', namespace, 'query'],
+          initialState.getIn(['namespaces', namespace, 'query'])
+        );
+    case SEARCH_BASE_QUERIES_UPDATE:
+      return state
+        .mergeIn(['namespaces', namespace, 'baseQuery'], baseQuery)
+        .mergeIn(['namespaces', namespace, 'query'], baseQuery)
+        .mergeIn(
+          ['namespaces', namespace, 'baseAggregationsQuery'],
+          baseAggregationsQuery
+        );
+    case SEARCH_QUERY_UPDATE:
+      return state
+        .mergeIn(
+          ['namespaces', namespace, 'query'],
+          state.getIn(['namespaces', namespace, 'baseQuery'])
+        )
+        .mergeIn(['namespaces', namespace, 'query'], query);
     case SEARCH_REQUEST:
-      return state.set('loading', true);
+      return state.setIn(['namespaces', namespace, 'loading'], true);
     case SEARCH_SUCCESS:
       return state
-        .set('loading', false)
-        .set('total', fromJS(action.payload.hits.total))
-        .set('sortOptions', fromJS(action.payload.sort_options))
-        .set('results', fromJS(action.payload.hits.hits))
-        .set('error', initialState.get('error'));
+        .setIn(['namespaces', namespace, 'loading'], false)
+        .setIn(['namespaces', namespace, 'total'], fromJS(data.hits.total))
+        .setIn(
+          ['namespaces', namespace, 'sortOptions'],
+          fromJS(data.sort_options)
+        )
+        .setIn(['namespaces', namespace, 'results'], fromJS(data.hits.hits))
+        .setIn(
+          ['namespaces', namespace, 'error'],
+          initialState.getIn(['namespaces', namespace, 'error'])
+        );
     case SEARCH_ERROR:
-      return state.set('loading', false).set('error', fromJS(action.payload));
+      return state
+        .setIn(['namespaces', namespace, 'loading'], false)
+        .setIn(['namespaces', namespace, 'error'], fromJS(error));
     case SEARCH_AGGREGATIONS_REQUEST:
-      return state.set('loadingAggregations', true);
-    case NEW_SEARCH_REQUEST:
-      return state.set(
-        'initialAggregations',
-        initialState.get('initialAggregations')
+      return state.setIn(
+        ['namespaces', namespace, 'loadingAggregations'],
+        true
       );
     case SEARCH_AGGREGATIONS_SUCCESS:
-      if (state.get('initialAggregations').isEmpty()) {
+      if (
+        state.getIn(['namespaces', namespace, 'initialAggregations']).isEmpty()
+      ) {
         // eslint-disable-next-line no-param-reassign
-        state = state.set(
-          'initialAggregations',
-          fromJS(action.payload.aggregations)
+        state = state.setIn(
```

Comment:
I don't get it, why are initialAggregations needed?

Reply:
it existed before. for now it's only need for literature date facet.